### DISCUSSION TITLE
the DOM won't update even signal was changed?

### DISCUSSION THREAD

**Discussion by maxpayne2046** [solidjs/solid#2044] [General]
![image](https://github.com/solidjs/solid/assets/85054690/bb4d4f9e-af62-495d-aca2-ece4f91b329b)

i am start learning SolidJS, I try many many times, but don't know while toggle che checkbox , the DOM won't update.

**Comment by atk** [comment]
The error is in toggleTodo. You're supposed to do it like this:

```js
setTodos(todos.findIndex(t => t.id === id), 'completed', (c) => !c);
```

**Reply by maxpayne2046** [reply]
thank you ,Alex. but i don't get the point.  the setTodos function can only accept 1 parameter,(either new value or callback func), right? i copy your code ,but I can't get it run. :-(

**Reply by maxpayne2046** [reply]
let copys = [...todos()];
    for (let todo of copys) {
      if (todo.id === id) {
        todo.completed = !todo.completed;
      }
    }
    setTodos(copys);


this one won't work ,neither.(i copy the values from getter to tmp , and modity it ,i print out to the console.log , tmp value is correct, but fail to pass to setter func? )

**Reply by atk** [reply]
Ah, I overlooked that you used createSignal instead of createStore. My bad.

**Reply by maxpayne2046** [reply]
ok,thanks Alex.I have no knowleadge about createStore, i am beginner. because createSignal can't track the change of deep nested objects,right?

**Reply by atk** [reply]
Exactly.

**Comment by maxpayne2046** [comment]
const toggleTodo = (id) => {
    let curTodos = todos();
    let i = curTodos.findIndex((t) => t.id === id);
    curTodos[i] = { id: id, text: curTodos[i].text, completed: !curTodos[i].completed };
    setTodos([...curTodos]);
  };

finally It works. have to remove old object and new one,can't simplely dirty the old data.  thanks alex.

**Reply by maxpayne2046** [reply]
i try createStore, next time,
